Tests/SwiftStdlibTests/StringExtensionsTests.swift

```diff
@@ -860,4 +878,31 @@ final class StringExtensionsTests: XCTestCase {
         XCTAssertNotNil(num.ordinalString())
         XCTAssertEqual(num.ordinalString(), "12th")
     }
+
+    func testReplacingOccurrencesPattern() {
+        let string = "hello"
+        XCTAssertEqual("hello", string.replacingOccurrences(of: "not", with: "found"))
+        XCTAssertEqual("hexo", string.replacingOccurrences(of: "l+", with: "x"))
+        XCTAssertEqual("hellxo", string.replacingOccurrences(of: "(ll)", with: "$1x"))
+
+        let range = string.startIndex..<string.index(string.startIndex, offsetBy: 3)
+        XCTAssertEqual("hexlo", string.replacingOccurrences(of: "l", with: "x", range: range))
```

Comment:
Could you add some bound off and empty string test cases?

Reply:
I'll add empty tests. Invalid range is hard to test because you can't even create them with `offsetBy:' - it throws an unchecked exception if you try `offsetBy: 99`. Not sure how to test that one, but I'm open to suggestions... Maybe we don't need it.

Reply:
Sure! I just suggested because normally in unit test we tend to think in add the most variable possibilities to make sure the expected behavior in most case we can :)

Reply:
No worries, I agree it's good practice and sometimes I need a nudge to avoid being lazy. 😄